src/components/Text/Text.tsx

```diff
@@ -13,20 +16,52 @@ export const RawText = forwardRef<TextProps, 'p'>((props, ref) => {
   return <CText sx={cTextStyle} ref={ref} {...props} />
 })
 
-export const Text = forwardRef<TextPropTypes, 'p'>((props, ref) => {
+export const Text = forwardRef<TextPropTypes, 'p'>(({ components, translation, ...props }, ref) => {
   const translate = useTranslate()
 
-  if (Array.isArray(props.translation)) {
+  const maybeElements = useMemo(() => {
+    if (!translation) return null
+
+    if (Array.isArray(translation) && !components) {
+      return (
+        <CText {...props} ref={ref}>
+          {translate(...translation)}
+        </CText>
+      )
+    }
+
+    if (!components) {
```

Comment:
ocd: `!components` is already checked for on the branch above. Either do explicit string checks on the left side here too (preferred), or use a single top-level `if (!components)` clause and two nested clauses with Array/string checks (also works but meh nested clauses)

This will improve the mental overhead of the branching 🙏🏽

Reply:
Done in https://github.com/shapeshift/web/pull/9075/commits/f7c9f4de0b138f8dd5005e524de395376f738b22